src/runtime/cpu_device_api.cc

```diff
@@ -32,8 +35,20 @@ class CPUDeviceAPI final : public DeviceAPI {
     ptr = memalign(alignment, nbytes);
     if (ptr == nullptr) throw std::bad_alloc();
 #else
-    int ret = posix_memalign(&ptr, alignment, nbytes);
-    if (ret != 0) throw std::bad_alloc();
+#ifndef _WIN32
+    constexpr size_t _HugePage2MB_ = 1<<21;
+    if (nbytes >= _HugePage2MB_) {
+       int ret = posix_memalign(&ptr, _HugePage2MB_, nbytes);
+       if (ret != 0) throw std::bad_alloc();
+       if ((ret= madvise(ptr, nbytes, MADV_HUGEPAGE )) != 0 )
+       throw std::bad_alloc();
+    } else {
+#endif
```

Comment:
`#endif  // _WIN32`
too many if/endif makes it hard to digest the code

Reply:
Agreed.  Seems to me that you don't have to condition on `_WIN32`, because on Windows either `_MSC_VER` (for MSVC) or `__MINGW32__` (for MinGW which we no longer supports) should be defined.